oasislmf/manager.py

```diff
@@ -440,26 +449,38 @@ def generate_oasis_files(
             target_dir,
             oasis_files_prefixes=files_prefixes['il']
         )
+        fm_summary_mapping = get_summary_mapping(il_inputs_df, hierarchy_terms)
+        write_mapping_file(fm_summary_mapping, target_dir)
 
         # Combine the GUL and IL input file paths into a single dict (for convenience)
         oasis_files = {**gul_input_files, **il_input_files}
 
+
         # If no RI input file paths (info. and scope) have been provided then
         # no RI input files are needed, just return the GUL and IL Oasis files
         if not (ri_info_fp or ri_scope_fp):
+            ## TODO: Write `fm_summary_map.csv`
+            # Write `summary_map.csv` for GUL+FM
             return oasis_files
 
         # Write the RI input files, and write the returned RI layer info. as a
         # file, which can be reused by the model runner (in the model execution
         # stage) to set the number of RI iterations
+
+
+        ## TODO: pass 'fm_summary_mapping` inplace of `xref_descriptions`
+        # Example Merge in col from exposure
+
+
+        xref_des = merge_oed_to_mapping(fm_summary_mapping, exposure_df, ['locgroup'])
```

Comment:
`LocGroup` (or `locgroup`) is an OED column in the locations file - I think it would be better, and consistent with existing practice, to use a profile to refer to any columns in the source files, e.g. a summary profile JSON, stored in the static data folder, which you can load via a wrapper method in `utils/defaults.py` (as with the existing locations and accounts profiles).